torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py

```diff
@@ -336,3 +296,99 @@ def aot_dispatch_autograd_graph(
             maybe_subclass_meta is None
         ), "aot_export_module does not support tensor subclass inputs for now."
     return fx_g, saved_updated_joint_inputs, maybe_subclass_meta
+
+
+def _aot_dispatch_function_wrapper(
+    fn: Callable[..., Any],
+    args: list[Tensor],
+    wrappers: Sequence[Callable[..., Any]],
+    *,
+    aot_config: AOTConfig,
+    functionalized_inputs: Optional[Sequence[Any]] = None,
+    **kwargs: Any,
+) -> tuple["FuncArgs", Optional[SubclassTracingInfo]]:
+    """Wrap a function with a list of tensor arguments"""
+    subclass_tracing_info: SubclassTracingInfo | None = None
+
+    @functools.wraps(aot_dispatch_subclass)
+    def _aot_dispatch_subclass(
+        fn: Callable[..., Any],
+        args: list[Tensor],
+        **kwargs: Any,
+    ) -> tuple[Callable[..., Any], list[Tensor]]:
+        # TODO(rec): make this purely functional
+        nonlocal subclass_tracing_info
+        subclass_tracing_info = aot_dispatch_subclass(fn, args, **kwargs)
+        return subclass_tracing_info[:2]
```

Comment:
what are the actual return invariants around each wrapper? (why do the other wrapper just return their output directly but not the subclass case? Is it because the subclass wrapper is unique in needing to return extra metadata?